### DISCUSSION TITLE
How to set K (DATA)

### DISCUSSION THREAD

**Discussion by Jazz9** [minio/minio#18696] [Q&A] [answered]
What is the configuration setting to set K ( DATA ) in the erasure coding?

**Comment by klauspost** [comment, marked as answer]
K is Set size minus erasure shards, so it is derived from other settings. See https://min.io/docs/minio/linux/operations/concepts/erasure-coding.html#erasure-coding-basics

**Comment by Jazz9** [comment]
Thankyou for the response.

Smaller K value should keep the part size higher and be helpful for HDD systems.

A 4K+2M should be twice as fast as a 10K+2M system at less than 1M S3 object sizes. Approx 1.5 time as fast at 2MB object size and the same speed above 2MB object size. With SSD the 10+2 would be quicker with more drives involved. 

Hard to pick a default! SSD is the future. We won't be building another HDD cluster.

**Reply by klauspost** [reply]
What you state depends on your use-case.

https://min.io/pricing
